docs/design.md

```diff
@@ -0,0 +1,785 @@
+defined by `parentSelector`, and the parent chosen for `region-1234` will be
+`global-1234` for the same reason. The rendered result for `site-1234` would
+be:
+
+```yaml
+---
+schema: example/Kind/v1
+metadata:
+  name: site-1234
+data:
+  a:
+    z: 3
+  b: 4
+...
+```
+
+If `region-1234` were later removed, then the parent chosen for `site-1234`
+would become `global-1234`, and the rendered result would become:
+
+```yaml
+---
+schema: example/Kind/v1
+metadata:
+  name: site-1234
+data:
+  a:
+    x: 1
+    y: 2
+  b: 4
+...
+```
+
+<!-- TODO: Add figures for this example, with region present, have site point
+with dotted line at global and indicate in caption (or something) that it's
+selected for but ignored, because there's a higher-precedence layer to select
+-->
+
+### Substitution
+
+Substitution is primarily designed as a mechanism for inserting secrets into
+configuration documents, but works for unencrypted source documents as well.
+Substitution is applied at each layer after all merge actions occur.
+
+Concrete (non-abstract) documents can be used as a source of substitution
+into other documents. This substitution is layer-independent, so given the 3
+layer example above, which includes `global`, `region` and `site` layers, a
+document in the `region` layer could insert data from a document in the
+`site` layer.
+
+Here is a sample set of documents demonstrating subistution:
+
+```yaml
+---
+schema: deckhand/Certificate/v1
+metadata:
+  name: example-cert
+  storagePolicy: cleartext
+  layeringDefinition:
+    layer: site
+data: |
+  CERTIFICATE DATA
+---
+schema: deckhand/CertificateKey/v1
+metadata:
+  name: example-key
+  storagePolicy: encrypted
+  layeringDefinition:
+    layer: site
+data: |
+  KEY DATA
+---
+schema: deckhand/Passphrase/v1
+metadata:
+  name: example-password
+  storagePolicy: encrypted
+  layeringDefinition:
+    layer: site
+data: my-secret-password
+---
+schema: armada/Chart/v1
+metadata:
+  name: example-chart-01
+  storagePolicy: cleartext
+  layeringDefinition:
+    layer: region
+  substitutions:
+    - dest:
+        path: .chart.values.tls.certificate
+      src:
+        schema: deckhand/Certificate/v1
+        name: example-cert
+        path: .
+    - dest:
+        path: .chart.values.tls.key
+      src:
+        schema: deckhand/CertificateKey/v1
+        name: example-key
+        path: .
+    - dest:
+        path: .chart.values.some_url
+        pattern: INSERT_[A-Z]+_HERE
+      src:
+        schema: deckhand/Passphrase/v1
+        name: example-password
+        path: .
+data:
+  chart:
+    details:
+      data: here
+    values:
+      some_url: http://admin:INSERT_PASSWORD_HERE@service-name:8080/v1
+...
+```
+
+The rendered document will look like:
+
+```yaml
+---
+schema: armada/Chart/v1
+metadata:
+  name: example-chart-01
+  storagePolicy: cleartext
+  layeringDefinition:
+    layer: region
+  substitutions:
```

Comment:
Great question.  @bryan-strassner and I had kicked around the idea of adding a new metadata schema for rendered documents that omitted this.  It might make sense for it to include additional details about what specific documents were involved in the layering and substitution process too. 
 What do you think about that general idea?

It's probably useful to return it on the non-rendered endpoint as it's part of the behavior definition.